src/lib/equinoxe/equinoxe.ml

```diff
@@ -70,36 +100,416 @@ module Make (B : Backend) : API with type 'a io = 'a B.io = struct
+        {
+          id = access "id" json |> Ezjsonm.get_string;
+          name = access "name" json |> Ezjsonm.get_string;
+          created_at =
+            access "created_at" json
+            |> Ezjsonm.get_string
+            |> Date.Parser.from_iso;
+          updated_at =
+            access "updated_at" json
+            |> Ezjsonm.get_string
+            |> Date.Parser.from_iso;
+        }
+      with Failure _ -> raise_wrong_date ~name:"Project.config_of_json" json
+
+    let to_string config =
+      let created_at = Date.Printer.to_iso config.created_at in
+      let updated_at = Date.Printer.to_iso config.updated_at in
+      Format.sprintf
+        "{\n\tid: %s;\n\tname: %s;\n\tcreated_at: %s;\n\tupdated_at: %s;\n}"
+        (replace_empty config.id)
+        (replace_empty config.name)
+        created_at updated_at
+
+    let get_all t =
+      let* json = Http.get ~t ~path:"projects" in
+      try
+        let projects =
+          access "projects" json |> Ezjsonm.get_list config_of_json
+        in
+        return projects
+      with Ezjsonm.Parse_error (v, err) ->
+        fail_with_parsing ~name:"Project.get_all" ~err ~json v
+
+    let get_from t ~id =
+      let path = Filename.concat "projects" id in
+      let* json = Http.get ~t ~path in
+      try return (config_of_json json)
+      with Ezjsonm.Parse_error (v, err) ->
+        fail_with_parsing ~name:"Project.get_from" ~err ~json v
+
+    let pp config = Format.printf "%s\n" (to_string config)
+  end
+
+  module State = struct
+    type t =
+      | Active
+      | Queued
+      | Provisioning
+      | Inactive
+      | Powering_off
+      | Powering_on
+
+    let to_string = function
+      | Active -> "active"
+      | Queued -> "queued"
+      | Provisioning -> "provisioning"
+      | Inactive -> "inactive"
+      | Powering_off -> "powering_off"
+      | Powering_on -> "powering_on"
+
+    let of_string = function
+      | "active" -> Active
+      | "queued" -> Queued
+      | "provisioning" -> Provisioning
+      | "inactive" -> Inactive
+      | "powering_off" -> Powering_off
+      | "powering_on" -> Powering_on
+      | s -> raise (Unknown_value ("State.of_string", s))
+  end
+
+  module Event = struct
+    type id = string
+
+    let id_of_string id = id
+    let id_to_string str = str
+
+    type t = {
+      id : id;
+      state : State.t;
+      event_type : string;
+      body : string;
+      created_at : Date.t;
+    }
+
+    let t_of_json json =
+      try
+        {
+          id = access "id" json |> Ezjsonm.get_string;
+          state = access "state" json |> Ezjsonm.get_string |> State.of_string;
+          event_type = access "type" json |> Ezjsonm.get_string;
+          body = access "body" json |> Ezjsonm.get_string;
+          created_at =
+            access "created_at" json
+            |> Ezjsonm.get_string
+            |> Date.Parser.from_iso;
+        }
+      with Failure _ -> raise_wrong_date ~name:"Event.t_of_json" json
```

Comment:
Perhaps we could add a wrapper around `Date.Parser.from_iso` to catch its `Failure` uniformly?